controllers/caregiver/caregiverController.js

```diff
@@ -75,3 +75,29 @@ export const getSingleCaregiverDetailController = async (req, res) => {
     res.status(500).json({ error: `An error occurred ${error}` });
   }
 };
+// Function to filter caregivers by date range
+export const filterCaregiversByDateController = async (req, res) => {
+  try {
+    const { startDate, endDate } = req.query; // Assuming startDate and endDate are query parameters
+
+    // Use Prisma to query caregivers whose availability falls within the selected date range
+    const caregivers = await withPrisma(async (prisma) => {
```

Comment:
this function should be moved to repositories folder, as it is going to be reusable by many controllers. so function that are related to DB query should be on repositories. as a result, controllers shouldn't bother importing withPrisma middleware.

I will fix this on master later so you can compare